### PR TITLE
Write MA not present to stdout

### PR DESCRIPTION
**Pull request by grk-it**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix
- [ ] New feature

### Testing

- [ ] Testing information has been added - test cases checklist and steps followed for testing.
- [ ] Testing screenshot(s) attached for more information.

## Security

- [ ] This PR has security related considerations.

### CODE REVIEW

**Comment by grk-it** [comment]
```
Fri Nov  8 06:21:31 2024 : Info: Adding client 127.0.0.1/32
Insecure packet from host 127.0.0.1:  Packet does not contain required Message-Authenticator attribute
```

File: src/lib/radius.c [hunk 1]
```diff
@@ -2813,10 +2813,11 @@ bool rad_packet_ok(RADIUS_PACKET *packet, int flags, decode_fail_t *reason)
 	    !packet->radiusv11 &&
 #endif
 	    !seen_ma) {
-		FR_DEBUG_STRERROR_PRINTF("Insecure packet from host %s:  Packet does not contain required Message-Authenticator attribute",
-			   inet_ntop(packet->src_ipaddr.af,
-				     &packet->src_ipaddr.ipaddr,
-				     host_ipaddr, sizeof(host_ipaddr)));
+                printf("Insecure packet from host %s:  Packet does not contain required Message-Authenticator attribute\n",
```
Comment (aren) [inline src/lib/radius.c:2816]: existing code to print bad shared secret uses `fr_strerror_printf()`

```
                        if (rad_digest_cmp(calc_auth_vector, msg_auth_vector,
                                   sizeof(calc_auth_vector)) != 0) {
                                fr_strerror_printf("Received packet from %s with invalid Message-Authenticator!  "
                                                   "(Shared secret is incorrect.)",
                                                   inet_ntop(packet->src_ipaddr.af,
                                                             &packet->src_ipaddr.ipaddr,
                                                             buffer, sizeof(buffer)));
                                /* Silently drop packet, according to RFC 3579 */
                                return -1;
```

**Review by aren — COMMENTED** [review]
also look at the function `blastradius_checks` in the upstream `v.3.2.x` branch which does a lot of these checks already.

File: src/lib/radius.c (hunk 1 above)
Comment (grk-it) [reply src/lib/radius.c:2816]: This didn't print the error on the cli. Looks like even  `Shared secret is incorrect.` gets printed only with `-X`

Comment (grk-it) [reply src/lib/radius.c:2816]: something funny is happening. On Madhav's branch the message `Shared secret is incorrect.` is not getting printed. But on `rel-2024-10` it gets printed.

Comment (grk-it) [reply src/lib/radius.c:2816]: please ignore the above comment. Something funny is happening with this particular error message ("I`nsecure packet from host 127.0.0.1:  Packet does not contain required Message-Authenticator attribute`"). `Shared secret is incorrect` gets printed without `-X` (I wasn't using radtest).

With `fr_strerror_printf` the error message is not printed. 
Screenshot 2024-11-08 at 10 06 32 AM

But with `printf` it gets printed.
Screenshot 2024-11-08 at 10 08 25 AM

Comment (grk-it) [reply src/lib/radius.c:2816]:
```
Fri Nov  8 19:43:16 2024 : Info: MainThread Found foxpass.com/test
Fri Nov  8 19:43:16 2024 : Info: MainThread (rid:none) virtual_client_response() completed in 0.04 seconds
Fri Nov  8 19:43:16 2024 : Info: Adding client 127.0.0.1/32
Fri Nov  8 19:43:16 2024 : Error: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Fri Nov  8 19:43:16 2024 : Error: BlastRADIUS check: Received packet without Proxy-State.
Fri Nov  8 19:43:16 2024 : Error: Setting "limit_proxy_state = true" for client foxpass.com
Fri Nov  8 19:43:16 2024 : Error: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Fri Nov  8 19:43:16 2024 : Error: The packet does not contain Message-Authenticator, which is a security issue.
Fri Nov  8 19:43:16 2024 : Error: UPGRADE THE CLIENT AS YOUR NETWORK MAY BE VULNERABLE TO THE BLASTRADIUS ATTACK.
Fri Nov  8 19:43:16 2024 : Error: Once the client is upgraded, set "require_message_authenticator = true" for client foxpass.com
Fri Nov  8 19:43:16 2024 : Error: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 INITIAL PACKET: {'request': {'User-Name': 'karthik', 'User-Password': 'test', 'Packet-Src-IP-Address': '127.0.0.1', 'Client-Shortname': 'foxpass.com', 'Tmp-String-1': '1', 'Tmp-String-0': 'initial'}, 'reply': {}, 'config': {}, 'session-state': {}, 'proxy-request': {}, 'proxy-reply': {}}
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 Initial packet from foxpass.com via 127.0.0.1, radsec client SN=None, radsec CA SN=None
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 (rid:1) initial_packet() completed in 0.02 seconds
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 (rid:1) {'User-Name': 'karthik', 'User-Password': '<< removed >>', 'Packet-Src-IP-Address': '127.0.0.1', 'Client-Shortname': 'foxpass.com', 'Tmp-String-1': '1', 'Tmp-String-0': 'outer-tunnel'}
Fri Nov  8 19:43:16 2024 : Error: Dummy-2 Error checking rate limit: Error 61 connecting to localhost:8321. Connection refused.
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 Allowing request to continue in spite of this error
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 Passwords match
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 returning OK, attributes None
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 (rid:1) outer_tunnel() completed in 0.25 seconds
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 post-auth outer-tunnel, eap-type None, post_auth_type: None
Fri Nov  8 19:43:16 2024 : Info: Dummy-2 (rid:1) request completed in 0.27 seconds
Fri Nov  8 19:43:25 2024 : Info: AsyncDBFlush Flushing 1 radclient updates
Fri Nov  8 19:43:25 2024 : Info: AsyncDBFlush Flushing 1 user RADIUS updates
```
regarding `blast_radius_check`: all the blast radius checks happen after virtual_client_response. But if MA required is set to Yes, blast radius checks don't emit any logs and the request is silently blocked.

**Pull request closed without merging**